Snowfork/polkadot-ethereum
# Relayer js test

instead of trying to fix the current relayer's reliability, i thought i'd try write a new simpler one.

it uses a stateless, idempotent, functional approach and is intended to be run, completed and then restarted every few seconds or so.

i'm fairly optimistic that this is probably the right thing to pursue for parachain -> ethereum channel relays, even after we add the data for MMRs and parachain merkle proofs. (though could continue in a golang script rather than JS)

thoughts? should we consider pursuing this path further?

to test it out:
 - checkout this branch
 -  run everything as usual (the original relayer is now limited to eth -> parachain mode only)
 - wait for everything to start, including parachain block production
 - run this new relayer with run.sh
 - run the e2e tests
 - watch the new relayer do its thing
 - all e2e tests should pass
